
    \thispagestyle{empty}
    
    \chapter[Why is selection for translationally optimal codons so scarce in metazoans? Variation in fitness effects and drift intensity]{Why is selection for translationally optimal codons so scarce in metazoans? Variation in fitness effects and drift intensity}
    \label{chap:CodonUsage}
    
    \vspace*{30pt}

    The third objective of my thesis is to study the variations in synonymous codon usage across metazoans. Synonymous codons do not modify the decoded amino acids. However, studies have shown that the different uses of these synonymous codons are not neutral and have an effect on the phenotype (\textit{i.e.} gene expression, translation \textit{etc.}).
    
    Interestingly, codon usage varies widely among metazoans and within genomes. There are two identified forces responsible for these variations. The first regroup non-adaptive processes such as gBGC or mutational bias, which affect both coding regions (\textit{i.e.} exons) and non-coding regions (\textit{i.e.} introns). Notably, in human, it has been observed that codon usage correlates with the GC content, and also that the GC content of bacterial genomes (from 13\% to 75\%) is strongly correlated with their CU. These results suggest that non-adaptive processes are at play in determining CU.
    
    The second process that drives codon usage is an adaptive processes, \textit{i.e.} translational selection, which favors the use of codons optimizing the speed and accuracy of translation, thereby affecting coding regions. In particular, this selection tends to promote in highly expressed genes the use of codons that match the tRNA pool, as seen in model species \textit{C. elegans}, \textit{D. melanogaster} and \textit{E. coli}.

    We aim to quantify translational selection across 257 metazoans, for which gene expression data are available in GTDrift. Our findings show that translational selection is rare in metazoans and its population-scaled selection coefficient ($S$) is low. In this range of $S$ values, the “drift barrier” suggests that reducing \Ne~leads to less efficient selection. Indeed, we observed low TS for low-\Ne~species. However, large-\Ne~species show a strong disparity in TS intensity. These variations are not simply explained by variations in mutational biases that could hamper TS. Thus, our results could suggest that the selective advantage in optimizing the translation machinery varies across species.

    \newpage
    
    \vspace*{10pt}
    
    \begin{center}
    \Large{\textbf{Why is selection for translationally optimal codons so scarce in metazoans? Variation in fitness effects and drift intensity}} \\
    \end{center}

    
    \begin{flushleft}
    \oneline{\large \textbf{{Florian Bénitière}$^{1,2}$ \orcidlink{0000-0001-7773-3542}, {Tristan Lefébure}$^{2}$ \orcidlink{0000-0003-3923-8166}, {Laurent Duret}$^{1}$ \orcidlink{0000-0003-2836-3463}}}
    
    \small
    \begin{center}
    $^{1}$Laboratoire de Biométrie et Biologie Évolutive, UMR CNRS 5558 LBBE, Université Claude Bernard Lyon 1, Villeurbanne, France\\
    $^{2}$Laboratoire d'Ecologie des Hydrosystèmes Naturels et Anthropisés, UMR CNRS 5023 LEHNA, Université Claude Bernard Lyon 1, Villeurbanne, France\\
    \end{center}
    \end{flushleft}

    {\hypersetup{linkcolor=GREYDARK}\minitoc}

    \newpage
    
\graphicspath{{chap7-Translational Selection/figures/}}



\section{Introduction}\label{sec:Introduction}

Since the early days of DNA sequencing, it has been noticed that the usage of synonymous codons is not random: some synonymous codons are more frequently used than others, and the patterns of synonymous codon usage (SCU) can vary both across species and among genes within a genome \citep{grantham_codon_1980-1}. Two types of processes, adaptive or non-adaptive, can contribute to genome-wide patterns of SCU \citep{sharp_codon_1993}. First, neutral substitution patterns (NSPs) vary across taxa and, in some species, can also vary along chromosomes. NSPs are primarily driven by the underlying pattern of mutation, which accounts for 60\% of the variance in genome base composition across the tree of life \citep{long_evolutionary_2018}. In addition, in some taxa, NSPs are also strongly affected by GC-biased gene conversion (gBGC), a process associated to homologous recombination that favors the transmission of G:C alleles over A:T alleles \citep{duret_biased_2009}. NSPs affect all genomic compartments (coding or non-coding), and notably have a strong impact on SCU (\textit{e.g.} \citet{pouyet_recombination_2017, long_evolutionary_2018}). Besides NSPs, SCU can also be affected by selection. Indeed, it has been observed that in some species, SCU varies according to gene expression level \citep{gouy_codon_1982, sharp_codon_1986, duret_expression_1999} and that the synonymous codons that are more frequently used in highly expressed genes correspond to the most abundant tRNAs \citep{ikemura_codon_1985, dong_co-variation_1996, moriyama_codon_1997, kanaya_studies_1999, duret_trna_2000}. This indicates that synonymous codon usage and tRNA content have coevolved in a way that optimizes translation. This co-evolution implies two levels of selection \citep{bulmer_coevolution_1987}: 1) selection on the pool of tRNAs to match the relative abundance of different codons in the transcriptome (\textit{i.e.} the codon demand), and 2) selection on the synonymous codon usage of genes to match the pool of tRNAs (classically referred to as “translational selection”). 

It is generally considered that there are two main benefits of using translationally optimal codons. First, this leads to increase the speed of translation, and hence to reduce the time spent by ribosomes on each mRNA, thereby increasing the pool of free ribosomes available in the cell, which ultimately allows a higher cellular growth rate \citep{bulmer_selection-mutation-drift_1991}. Second, the usage of synonymous codons decoded by the most abundant tRNAs increases the accuracy of translation, and thus reduces the amount of mis-translated proteins that cause an important burden on the cell \citep{akashi_synonymous_1994, drummond_single_2006}. It is important to note that for both aspects (speed and accuracy of translation), the benefit of using optimal codons is expected to be proportional to gene expression level. Indeed, the higher the expression level of a given gene, the stronger the impact of its translation speed on the pool of free ribosomes, and for a given mis-translation rate, the cost of erroneous protein production (in terms of waste of resources and of direct toxic effect of misfolded proteins) increases directly with expression level. In bacteria, the intensity of translational selection is correlated to the minimal cell division time, which suggests that the selective force for the optimization of SCU is the maximization of cellular growth \citep{rocha_codon_2004, sharp_variation_2005}.

It should be noted that besides translational selection, synonymous sites can be subject to additional levels of selective constraints. For instance, the presence of splice enhancers located within exons skews codon usage near exon-intron boundaries in mammalian genes \citep{parmley_exonic_2007}. But this type of selective pressure is site-specific (\textit{i.e.} a particular codon is preferred at a specific site in a given gene), and hence, is not expected to affect the genome-wide pattern of SCU. Similarly, there is evidence that the use of translationally sub-optimal codons can be advantageous at some specific sites to slow-down translation and favor the proper folding of proteins \citep{buhr_synonymous_2016, walsh_synonymous_2020}. But again, this is a local effect, with limited genome-wide impact on SCU.

Interestingly, the intensity of translational selection varies widely across species, not only in unicellular organisms, but also in multicellular eukaryotes \citep{sharp_variation_2005, subramanian_nearly_2008, dos_reis_estimating_2009, galtier_codon_2018}. For instance, among animals, early studies on the two main invertebrate model organisms (\textit{Drosophila melanogaster} and the nematode \textit{Caenorhabditis elegans}) showed clear signatures of translational selection \citep{shields_silent_1988, duret_expression_1999}. Conversely, there is no sign of translational selection in humans \citep{semon_no_2006, pouyet_recombination_2017}, despite clear evidence that SCU does affect gene expression in mammals \citep{kudla_high_2006, courel_gc_2019, wu_translation_2019, mordstein_codon_2020, medina-munoz_crosstalk_2021}. To understand variation in the intensity in translational selection across animals, it is important to refer to basic population genetics principles \citep{ohta_neutral_1996}. Indeed, the SCU in a given genome reflects a balance between selection favoring translationally optimal codons, and the effects of mutation and drift, allowing the fixation of non-optimal codons \citep{bulmer_selection-mutation-drift_1991}. Thus, the frequency of optimal codons is expected to depend on the population-scaled selection coefficient ($S=4\textit{N}_{\text{e}}s$), where \textit{N}$_{\text{e}}$ is the effective population size and $s$ the selection coefficient in favor of translationally optimal codons \citep{bulmer_selection-mutation-drift_1991, sharp_variation_2005}. Hence, the lack of translational selection in some animal taxa might stem from a small \textit{N}$_{\text{e}}$ (hereafter referred to as the drift-barrier hypothesis), or from a smaller fitness effect of using translationally optimal codons (\textit{i.e.} lower $s$).

To explore these hypotheses, several previous studies analyzed variation in the intensity of translational selection across eukaryotes \citep{subramanian_nearly_2008, dos_reis_estimating_2009, galtier_codon_2018}. These three studies, reported positive correlations between signatures of translational selection and proxies of \textit{N}$_{\text{e}}$ \citep{subramanian_nearly_2008, dos_reis_estimating_2009, galtier_codon_2018}. Although this pattern fits qualitatively with the predictions of the drift barrier model, quantitatively, the fit is not so clear. Indeed, \citet{dos_reis_estimating_2009} estimated $S$ in 10 eukaryotic species, and they reported only a 2-fold difference in $S$ between humans and \textit{D. melanogaster} (respectively $S=0.5$ and $S=1.0$), despite a $\approx$30-fold difference in \textit{N}$_{\text{e}}$ between the two species (20,000 vs. 600,000; \citet{lynch_divergence_2023}). According to the authors, this poor fit to the drift barrier model might be due to the fact that their analysis was sensitive to variation in NSP across genes, which might have led to overestimate $S$ in humans \citep{dos_reis_estimating_2009}. But, it has also been argued that besides differences in \textit{N}$_{\text{e}}$, $s$ is also likely to vary across species, as long-lived organisms, with relatively a slow development, are likely to be less constrained to optimize cell growth than species with a very rapid development \citep{subramanian_nearly_2008}. 

These three studies were based on relatively limited sample sizes (10 to 30 species), and in the end, the causes of the variation in the intensity of translational selection across species remained unclear. To try to go further, we decided here to investigate variation in translational selection intensity across a large dataset of 223 metazoan species, covering a wide range of animal clades. For each species, we predicted the set of optimal codons based on the pool of tRNA genes present in its genome, and we analyzed how the frequency of optimal codons varies with gene expression, controlling for variation in NSP. Based on these variations, we quantified $S$ in each species, and analyzed how it correlates with estimates of \textit{N}$_{\text{e}}$ or life history traits. Our analyses revealed that overall, few metazoans show clear signs of translational selection. As expected, the highest values of $S$ are observed in species with large \textit{N}$_{\text{e}}$, while species with small \textit{N}$_{\text{e}}$ show little evidence of translational selection. However, overall, \textit{N}$_{\text{e}}$ appears to be a poor predictor of the intensity of translational selection, which suggests important variation in $s$ across taxa. We discuss several factors that may drive this variation in the fitness effect of optimizing codon usage. 



\section{Results}\label{sec:Results}

\subsection{Non-adaptive processes are the primary drivers of codon usage variations among metazoans}
\label{subsec:nonadapt}

To investigate the factors driving the intensity of translational selection in metazoans, we used the GTDrift database, that compiles genomic and transcriptomic data along with life history traits and proxies of \textit{N}$_{\text{e}}$ for various eukaryotic species ~\citep{benitiere_gtdrift_2024}. We initially selected 257 metazoan species available in GTDrift, but we excluded 11 species for which there were not enough transcriptomic data (less than 5,000 genes detected as being expressed). We analyzed patterns of SCU and genomic base composition in the 246 remaining species, covering a wide range of clades (129 vertebrates, 82 insects and 35 other metazoan species; \hyperref[fig:CU1]{Fig. 1A}). 

Patterns of SCU can be affected both by translational selection and by NSPs \citep{sharp_codon_1993}. It is possible to distinguish the contribution of NSPs because they affect the base composition of both coding and non-coding regions, whereas translational selection operates only on codons. Thus, if differences in SCU across species are driven by NSPs, then it is expected that they should correlated with variation in the base composition of non-coding regions. And similarly, if intra-genomic variation of SCU in a given species is driven by the heterogeneity of NSPs along its chromosomes, then this should result in a covariation between the codon usage of genes and the base composition of their introns. Owing to the symmetry of the DNA molecule, NSPs generally affect similarly both strands, resulting in an equal proportion of cytosine (C) and guanine (G), as well as an equal proportion of thymine (T) and adenine (A) \citep{lobry_properties_1995}. Hence, the G+C content provides a good summary statistics of the impact of NSPs on the genomic base composition. Thus, to examine the potential contribution of non-adaptive processes to the observed variations in SCU across the 246 species, we measured their G+C content in introns (GCi) and at the third position of codons (GC3), averaged over all genes. We observed a strong correlation between the average GC3 and the average GCi (\hyperref[fig:CU1]{Fig. 1B}). Our findings suggest that non-adaptive processes, are the primary factor driving the observed variation in codon usage across species. As already noted by \citet{vinogradov_dna_2003} and \citet{amit_differential_2012}, the relationship between GC3 and GCi is asymmetrical. While introns are predominantly AT-rich (GCi range=0.2 to 0.52), the third position of codons displays a wider range of variation, with GC3 spanning from 0.25 to 0.73. While most species of a clade displayed similar average GC3, dipterans (N=20) exhibit the widest range of GC3 variations (from 0.32 to 0.71).

\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=\textwidth]{Figure1.pdf}
    \end{center}                                                                       
    \caption[Codon usage variations are driven by non-adaptive processes]{\textbf{Codon usage variations are driven by non-adaptive processes.} \textbf{A}: Phylogenetic tree of the 257 studied species. \textbf{B}: Gene average GC content at the third position of codons (GC3) and the gene average GC in introns (GCi) for each species. Pagel's \textit{lambda} model is used to take into account the phylogenetic structure of the data in a regression model (black line). \textbf{C}: Correlation between the gene GC3 and the GCi in \textit{Homo sapiens} (left) and in \textit{Caenorhabditis elegans} (right). Spearman's rho and corresponding p-values are displayed under the graph. The dotted lines correspond to x=y.}
    \label{fig:CU1}
\end{figure*}

NSPs can vary within the genome of a given species, and impact codon usage accordingly. In \textit{Homo sapiens}, the \textit{per} gene GC3 and GCi are highly correlated (Spearman's correlation coefficient, rho=0.83, p$<$$10^{-16}$), whereas this correlation is less pronounced in \textit{Caenorhabditis elegans} (rho=0.24, p$<$$10^{-16}$; \hyperref[fig:CU1]{Fig. 1C}). Species showing the strongest intra-genomic variance in codon usage (as assessed by  GC3) are the ones with the strongest variance in GCi \hyperref[suppfig:CU1]{Supplementary Fig. 1A}). These correlations between GC3 and GCi are particularly strong in tetrapods (107/108 species with rho$>$0.7) and in hymenopterans (25/35 species with rho$>$0.7)  (\hyperref[suppfig:CU1]{Supplementary Fig. 1B}). The other clades generally show less variance in GCi, and weaker correlations between GC3 and GCi. But overall, 244/246 species (99\%) showed a significant positive correlation (p$<$0.05), which indicates that in most species, intra-genomic variation in NSPs somehow contribute to the variance in SCU among genes. Hence, it is important to take this source of variance into account to be able to detect signatures of translational selection within genomes.

\subsection{tRNA abundance matches proteome requirements}

To quantify the intensity of translational selection, we used an approach similar to that of \citet{dos_reis_estimating_2009} and \citet{sharp_variation_2005}. This approach is based on the comparison of the frequency of optimal codons between highly and weakly expressed genes, and therefore requires the prior identification of optimal codons. For this, dos Reis and Wernisch focused on the nine amino-acids that are encoded by two codons (duet codons), and predicted the optimal codon of each amino-acid as being the one that is more frequently used in highly expressed genes. One caveat is that if the NSP varies among genes according to their expression level, this may lead to erroneous prediction of codon optimality. Furthermore, this approach does not capture the signal of translational selection from the nine other amino-acids that are encoded by triplet, quartet or sextet codons. To avoid these limitations, we sought here to predict optimal codons based on the tRNA pool. Owing to technical difficulties, there are currently few species for which tRNA abundance has been quantified directly. \citet{behrens_high-resolution_2021} recently developed a technique (mim-tRNAseq) that allowed them to measure tRNA abundance in four eukaryotes \citep{behrens_high-resolution_2021}. This study revealed a robust correlation between tRNA abundance and their respective gene copy number, with an adjusted $R^2 > 0.91$ for yeasts (\textit{S. cerevisiae} and \textit{S. pombe}), 0.79 for \textit{Drosophila melanogaster} and 0.62 for \textit{Homo sapiens}~\citep{behrens_high-resolution_2021}. These results suggest that tRNA copy numbers are a good predictor of tRNA abundances. To investigate whether the number of tRNA genes could be used as an indirect measure of tRNA abundance across metazoans, we analyzed the co-variation of their tRNA gene repertoires with the amino acid composition of their proteome.

The total number of tRNA gene copies varies widely among clades and species (ranging from an average of 201 tRNA gene copies per genome in hymenopterans to 1,537 copies in teleost fish; \hyperref[suppfig:CU2]{Supplementary Fig. 2}). However, the relative copy number of distinct isoacceptor tRNA genes is quite conserved among metazoans. There are some rare cases where the gene copy number of a given tRNA has exploded in a given species compared to other genomes (\hyperref[suppfig:CU2]{Supplementary Fig. 2}). This might reflect the propensity of tRNA genes to become transposable elements. Indeed many SINE retrotransposon families derive from tRNA genes \citep{sun_common_2007}, and it is therefore possible that some recently evolved SINEs are erroneously annotated as \textit{bona fide} tRNA genes.

In both \textit{Drosophila melanogaster} and \textit{Homo sapiens}, we observed a strong correlation between amino acid usage (\textit{i.e.} the frequency of amino acids, weighted by the expression level of genes) and direct measures of tRNA abundance (rho=0.79; \hyperref[suppfig:CU3]{Supplementary Fig. 3}; \citet{behrens_high-resolution_2021}). These results indicate that tRNA abundance matches the amino acid demand. As expected, the amino acid usage of these two species also strongly correlates with their tRNA gene copy numbers (rho=0.78 and 0.68 respectively; \hyperref[suppfig:CU3]{Supplementary Fig. 3}). As previously reported \citep{duret_trna_2000}, tRNA gene copy number also correlates with the amino acid demand in \textit{Caenorhabditis elegans} (rho=0.82; \hyperref[fig:CU2]{Fig. 2A}). The same analysis conducted across 246 animal species found a significantly positive Spearmann coefficient (\textit{i.e.} p-value $<$ 0.05) in 93\% of the species (\hyperref[fig:CU2]{Fig. 2B}), which indicates that in most of metazoans, the tRNA gene copy number is under constraints to match the amino acid demand. This implies that tRNA abundance is primarily regulated by modulating the copy number of tRNA genes rather than their transcription level. We suspect that the few cases where the number of tRNA genes does not correlate with amino acid usage might be due to annotation errors : some tRNA genes may have been missed (\textit{e.g.} because of gaps in the genome assembly), or conversely, some SINEs or tRNA pseudogenes may have been incorrectly annotated as functional tRNA genes. To ensure that the tRNA gene copy number is a good proxy of the tRNA abundance, we kept in our study only the species for which tRNA gene copy number correlates significantly with amino acid usage (N=230 species). We also excluded 7 species for which the repertoire of annotated tRNA appeared to be incomplete (\textit{i.e.} the cognate tRNAs of certain codons were not found in the genome assembly).

\begin{figure*}[t]
    \begin{center}
        \includegraphics[width=\textwidth]{Figure2.pdf}
    \end{center}                                                                       
    \caption[The tRNA gene copies number is a good predictor of the transcriptional requirements]{\textbf{The tRNA gene copies number is a good predictor of the transcriptional requirements.}
    \textbf{A}: The relationship between the number of tRNA gene copies \textit{per} amino acid and the frequency of amino acid weighted by gene expression (FPKM, log scale) in \textit{Caenorhabditis elegans}. Spearman's rho and corresponding p-value are displayed under the graph. \textbf{B}: Boxplot illustrating the distribution of Spearman's correlation coefficient ($\rho$) from Panel A for each species (N=246 species). The red line indicates the threshold above which the p-value is lower than 0.05.}
    \label{fig:CU2}
\end{figure*}


\subsection{Definition of putative-optimal codons based on tRNA abundance and wobble-pairing rules}

To predict which synonymous codons are optimal for translation, it is first necessary to associate each of the 61 codons to their cognate tRNA. The number of distinct isodecoder tRNAs (\textit{i.e.} distinct anticodons) ranges from 43 to 60 per species (average=47). This implies that 1 to 18 codons cannot be translated through Watson-Crick pairing (WCp), and hence have to be translated via wobble pairing (WBp). We used the rules established by Percudani \citep{percudani_restricted_2001} to assign each of these codons to their cognate tRNA, allowing for non-standard base pairing with the first nucleotide of the anticodon (\hyperref[fig:CU3]{Fig. 3A}). For example, deamination of adenine in inosine (I) in anticodons ANN makes them permissive to wobble pairing I:C; I:U or I:A. Another common wobble pairing is the G:U/U:G pairing~\citep{percudani_restricted_2001}. As an illustration, in human, asparagine is translated by a single tRNA (anticodon GTT) that decodes both AAC (by WCp) and AAT (by G:U WBp). AAT accounts for 48\% of asparagine codons, highlighting the significance of wobble pairing. 
There are 18 amino acids that are encoded by multiple synonymous codons. These amino acids can be classified in two groups:
- those whose synonymous codons are translated by at least two distinct isodecoder tRNAs 
- those for which all synonymous codons are translated by a single isodecoder tRNA 

There is some variation in the set of amino acids present in each group, depending on the isodecoder tRNA repertoire of each species. The first group generally corresponds to amino acids encoded by sextet codons (Leu, Arg, Ser), quartets (Val, Gly, Ala, Pro, Thr), triplet (Ile) and NNG/NNA duets (Glu, Gln, Lys). The second group corresponds essentially to the six amino acids encoded by NNC/NNT duets (Phe, Cys, Tyr, Asp, His, Asn) (\hyperref[fig:CU3]{Fig. 3B}).

For each amino acid of the first set, synonymous codons were predicted to be optimal if they were decoded by the isodecoder tRNA with highest gene copy number (\textit{i.e.} predicted to be the most abundant). In case of ex æquo (\textit{i.e.} if all synonymous codons are decoded by isodecoders having the same gene copy number), then the optimal codons of this amino acid were considered as unknown (76 ex æquo cases in total, in 62 species). In the cases where the most abundant tRNA decodes more than one synonymous codon, we considered all of them as potentially optimal (\textit{i.e.} at this stage, we do not make any assumption regarding which of the Watson-Crick pairing or wobble pairing is the most efficient). This first set of putative-optimal codons will hereafter be referred as 'POC1'.
The second set of amino acids corresponds to cases where the two synonymous codons (NNC/NNT) are decoded by a single isodecoder (anticodon GNN). There is evidence, based on studies in various eukaryotes, that the wobble pairing GNN:NNU is less efficient than the Watson-Crick pairing GNN:NNC~\citep{stadler_wobble_2011, chan_frequent_2017, wang_accurate_2017}. Consequently, for these amino acids, we defined codons NNC (decoded through WCp) as being the putative-optimal codons 'POC2'.

For the human genome, POC1 have been defined for 13 amino acids and POC2 for 5 amino acids. In contrast, for \textit{Caenorhabditis elegans}, POC1 and POC2 are defined for 12 and 6 amino acids, respectively. On average among the 223 species, POC1 are defined for 12.5 amino acids \textit{per} species (ranging from 8 to 17) and POC2 for 5.2 amino acids (ranging from 0 to 6, except \textit{Tyto alba} with 7 POC2, including Ile).

\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=\textwidth]{Figure3.pdf}
    \end{center}                                                                       
    \caption[Presence-Absence of tRNA defines set of putative-optimal codons]{\textbf{Presence-Absence of tRNA defines set of putative-optimal codons.} \textbf{A}: Illustration of the various possible pairings: Watson-Crick and wobble pairing. \textbf{B}: A boxplot illustrating the distribution of tRNA gene copy numbers across 223 species. The percentage of species lacking a tRNA gene copy is also indicated, highlighting the absence of tRNA isodecoder.}
    \label{fig:CU3}
\end{figure*}


\subsection{Highly expressed genes are enriched in optimal codons}

The intensity of translational selection depends directly on gene expression levels. Given the very wide range of variation of gene expression levels ($> 1000$ folds), the fitness impact of synonymous codon usage is expected to vary strongly among genes. Hence, a typical feature of genomes subject to translational selection (TS), is that the frequency of optimal codons is particularly high in the most highly expressed genes. Thus, to identify which species are subject to TS, we examined the variations in POC frequency according to gene expression level. To control for possible variations in neutral substitutions pattern, we also analyzed triplet content in introns, referred to as POC-control. It is important to note that the frequency of POC-control is not expected to be equivalent to that of the overall POC frequency due to the differing AT-richness of introns.

For \textit{Homo sapiens} POC frequencies show some slight fluctuations according to gene expression level (\hyperref[fig:CU4]{Fig. 4A}). However, the same weak fluctuations are observed for POC-controls in introns, which implies that a same process, independent of translation efficiency, affects the base composition, both in introns and at synonymous codon positions (\hyperref[fig:CU4]{Fig. 4A}). Indeed, there exists a strong correlation between the GC content of introns and the GC3 content in \textit{Homo sapiens}, along with pronounced variations in GC3 content (\hyperref[fig:CU1]{Fig. 1C}). In contrast, in \textit{Caenorhabditis elegans}, we observed a strong rise in POC frequencies in highly expressed genes, both for POC1 (from 47\% to 76\%) and for POC2 (from 38\% to 70\% ). These changes in codon usage are not caused by shift in local substitution patterns as we see no similar variation in POC-control (\hyperref[fig:CU4]{Fig. 4B}).


It is important to notice that the non-linear relationship observed between gene expression level and POC frequency is perfectly consistent with the TS model, that assumes that the selection coefficient on synonymous codon usage ($s$) should increase linearly with gene expression level. Indeed, this model predicts that for lowly expressed genes (such that $S = 4$\textit{N}$_{\text{e}}s \ll 1$), the frequency of optimal codons should evolve neutrally, and hence should be independent of expression level. But above the 'nearly-neutral' point (\textit{i.e.} the expression level for which $S\approx1$), the frequency of optimal codon should strongly increase with expression level (see \hyperref[appendices]{Appendices} for more details on equations). The shape of the POC1 and POC2 curves in \hyperref[fig:CU4]{Fig. 4B} indicates that in \textit{C. elegans}, this 'nearly-neutral' point is reached for a gene expression level of about 50 FPKM. This implies that genes with a lower expression level (which represent 83\% of genes in \textit{C. elegans}) are not affected by TS. Of note, the fraction of genes affected by TS is expected to be even more reduced in species with a lower effective population size.

To assess the impact of TS on synonymous codon usage, we measured the difference between the frequency of POCs in the most expressed genes (top 2\%), and the frequency of POCs in the 50\% lowest expressed genes, controlling for POCs-control variations (see \nameref{sec:MaterialsMethodsCU}). This shift in codon usage (denoted $\Delta POC^{exp}$ ) was computed for both POC1 and POC2 codons in each of the studied species (N=223 species). $\Delta POC_1^{exp}$ and $\Delta POC_2^{exp}$ are strongly correlated ($R^2=46$\% , p-value $< 10^{-16}$), which indicates that the signature of translational selection is effectively captured by both sets of codons (\hyperref[fig:CU4]{Fig. 4C}). For 211 species (95\%), the prevalence of POC1 is greater in highly expressed genes compared to other genes (\hyperref[fig:CU4]{Fig. 4D}). Similarly, $\Delta POC_2^{exp}$ is positive for 191 species (86\%; \hyperref[fig:CU4]{Fig. 4D}). The highest values of $\Delta POC^{exp}$ were observed in \textit{C. elegans} (+30\% for both POC1 and POC2). There are substantial variation across clades: average $\Delta POC_1^{exp}$ and $\Delta POC_2^{exp}$ values are around +14\% in Diptera compared to +3\% in vertebrates. We obtained very similar results when measuring $\Delta POC^{exp}$ without accounting for POC-control variations (\hyperref[suppfig:CU5]{Supplementary Fig. 5}). Given that the two sets of codons gave very consistent results, we hereafter considered the whole set of POCs regrouping both POC1 and POC2 .


\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=\textwidth]{Figure4.pdf}
    \end{center}                                                                       
    \caption[Differences in usage of putative-optimal codon between highly- and lowly-expressed genes]{\textbf{Differences in usage of putative-optimal codon between highly- and lowly-expressed genes.} \textbf{A,B}: Variation in the proportion of POC within coding sequences (POC1: dark blue; POC2: dark green) according to gene expression level. To control for variations in neutral substitution patterns, we analyzed the frequency of corresponding triplets within introns (POC1 control: light blue; POC2 control: light green). Each point represents a 2\% bin of genes, with the red point at the end of each POC1 curve denoting the 2\% most highly expressed genes. The red lines indicate the average POC1 proportions observed in the 50\% least expressed genes (FPKM, log scale). \textbf{A} represents \textit{Homo sapiens} , and \textbf{B} represents \textit{Caenorabditis elegans}. \textbf{C}: Relation between the variations in POC1 and POC2 frequency with expression. Calculated as the difference between POC frequency in the 2\% most highly expressed genes and the 50\% least expressed genes. To this variations we removed variations measured on control. \textbf{D}: Boxplot illustrating the differences, for each species, between the TS intensity measured on POC1 and POC2.
    }
    \label{fig:CU4}
\end{figure*}


\subsection{Highly constrained amino acids are enriched in optimals codons}

Synonymous codon usage is expected to be under selection not only for its impact on the speed of translation, but also on the accuracy of translation. For both traits, selective constraints are expected to vary among genes, according to their expression level. One specific feature of selection for translation accuracy is that the strength of selection is also expected to vary among sites within a protein: selection on translation accuracy should be stronger at sites that are essential for the structure or function of the protein. 
To test this prediction, we analyzed within-gene variation in POC usage according to the level of constraint on amino acid sites. For this, we focused on a set of 976 orthologous genes, present in single copy in most metazoan genomes (BUSCO genes; \citet{waterhouse_busco_2018}). For each protein of a given species, we classified its sites into four groups of equal size, according to their level of conservation across 293 metazoans (see \nameref{sec:MaterialsMethodsCU}), and then measured the shift in POC frequency between its 25\% most conserved sites and its 25\% least conserved sites. Finally, we computed the average of these shift values over all proteins of this species (noted $\Delta POC^{cons}$). Given that the shift is computed within each gene, $\Delta POC^{cons}$ measures variation in codon usage across sites that inherently have the same expression level. 
One difficulty however is that in tetrapods, many genes contain a GC-rich CpG island at their 5' end \citep{deaton_cpg_2011}. The presence of a CpG island affects the base composition of the beginning of genes, up to about 1 kb, as illustrated by the analysis of the intronic GC content (\hyperref[suppfig:CU7]{Supplementary Fig. 7}). This results in differences in codon usage between the first exon and the rest of the coding region. Given that the N-termini of proteins evolve faster than their center \citep{bricout_evolution_2023}, this causes a spurious association between codon usage and variation in amino acid constraints along proteins. To avoid this bias, we measured $\Delta POC^{cons}$ in tetrapods only on codons located beyond 1 kb of the start codon (in genomic coordinates). In other clades, the base composition of introns shows little variation along genes (\hyperref[suppfig:CU7]{Supplementary Fig. 7}), and hence $\Delta POC^{cons}$ was measured on the entire coding region. 
In \textit{C. elegans}, the frequency of POC increased significantly between the least constrained and most constrained sites within proteins (from 48.5\% on average to average 51.2\%), whereas no variation was observed in humans (\hyperref[fig:CU5]{Fig. 5A,B}). Overall, $\Delta POC^{cons}$ is positive in 75\% of species (refer to \hyperref[fig:CU5]{Fig. 5C}). As for $\Delta POC^{exp}$ , $\Delta POC^{cons}$ shows substantial variation across clades, and is maximal for Diptera.



\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=0.75\textwidth]{Figure5.pdf}
    \end{center}                                                                       
    \caption[Most highly conserved regions exhibit a preference for using POCs]{\textbf{Most highly conserved regions exhibit a preference for using POCs.} \textbf{A,B}: Investigation of POCs frequency by dividing genes into four constraint groups of equal size, based on the gap proportion of gene alignments across all species. The frequency of POCs was calculated for each gene within each constraint group. A boxplot is shown, with darker green indicating the least constrained sites and darker red indicating the most constrained sites. \textbf{A} represents \textit{Homo sapiens}, and \textbf{B} represents \textit{Caenorabditis elegans}. \textbf{C}: Distributions depicting the average differences \textit{per} species between the frequency of POCs in highly constrained sites and unconstrained sites of given genes.}
    \label{fig:CU5}
\end{figure*}


\subsection{Selection favors optimal codons in highly expressed genes of \textit{Drosophila melanogaster}}

To further assess whether POCs are under selection in Diptera, we investigated patterns of polymorphism and substitution in \textit{Drosophila}, based on a multiple genome alignment of three closely related species (\textit{D. melanogaster, D. simulans, D. erecta}) and on single nucleotide polymorphism (SNP) data from 205 \textit{D. melanogaster} individuals. We inferred the ancestral and derived state at each substitution or SNP, so that to distinguish synonymous changes corresponding to POC to non-POC mutations (PO$>$nPO) vs non-POC to POC mutations (nPO$>$PO) (see \nameref{sec:MaterialsMethodsCU}; \hyperref[fig:CU6]{Fig. 6A}). To control for possible variation in local mutation patterns, we conducted a parallel analysis on triplets in intronic regions. In coding sequences (CDS), we identified 44,288 nPO$>$PO synonymous SNPs and 139,256 PO$>$nPO synonymous SNPs, 26,770 nPO$>$PO synonymous substitutions and 81,666 PO$>$nPO synonymous substitutions. In introns, we observed 187,321 nPO$>$PO SNPs and 260,366 PO$>$nPO SNPs, 56,916 nPO$>$PO substitutions and 77,686 PO$>$nPO substitutions. 

We observed that the rate of nPO$>$PO changes (number of nPO$>$PO changes/number of non-POC codons) increases with increasing gene expression level, while the rate of PO$>$nPO changes (number of PO$>$nPO changes/number of POC codons) decreased, both for SNPs and for substitutions(\hyperref[fig:CU6]{Fig. 6B,D}). Importantly, this trends is specific to coding regions, and is not observed for the corresponding triplets in introns (\hyperref[fig:CU6]{Fig. 6C,E}). These observations are consistent with the hypothesis that selection favors mutations leading to the incorporation of translationnally optimal codons in genes with high expression level.



\begin{figure*}[t]
    \begin{center}
        \includegraphics[width=0.9\textwidth]{Figure6.pdf}
    \end{center}                                                                       
    \caption[Selective pressure on non-POCs to POCs mutations]{\textbf{Selective pressure on non-POCs to POCs mutations.} \textbf{A}: Schematic representation of the method used to identify SNPs and substitutions in \textit{Drosophila melanogaster}. 
    \textbf{B,C}: Rate variations of SNPs non-POC towards POC (green) and POC towards non-POC (red) with gene expression in CDS (\textbf{B}) and in intronic control (\textbf{C}). 
    \textbf{D,E}: Rate variations of substitutions non-POC towards POC (green) and POC towards non-POC (red) with gene expression in CDS (\textbf{D}) and in intronic control (\textbf{E}). Error bars represent the 2.5th and 97.5th percentiles of values obtained from 100 simulations using a binomial distribution, following the same site structure and substitution rate (see \nameref{sec:MaterialsMethodsCU}).}
    \label{fig:CU6}
\end{figure*}

\subsection{Weak relationship between the strength of translational selection and the effective population size}

According to standard population genetic models of translational selection (\citet{bulmer_selection-mutation-drift_1991, sharp_variation_2005, dos_reis_estimating_2009}; \hyperref[appendices]{Appendices}), the difference in codon usage between highly and weakly expressed genes is expected to be directly linked to the population-scaled selection coefficient in favor of optimal synonymous codons ($S=4\textit{N}_{\text{e}}s$). Indeed, considering that synonymous codon usage evolves neutrally in lowly expressed genes, then $S$ in highly expressed genes can be expressed as: \begin{equation}S^{hx}=ln(\frac{FOP^{hx}}{1-FOP^{hx}})-ln(\frac{FOP^{lx}}{1-FOP^{lx}})\end{equation} where $FOP^{hx}$ and $FOP^{lx}$ are the observed frequencies of optimal codons in highly and lowly expressed genes respectively \citep{sharp_variation_2005, dos_reis_estimating_2009}. It should be noted however that this equation holds true only if underlying mutation patterns (and possibly gBGC) do not vary with gene expression level \citep{sharp_variation_2005, dos_reis_estimating_2009}. We used the above equation to estimate $S^{hx}$ in each species, based on the observed POC frequencies in the top 2\% most highly expressed genes, compared to the 50\% least expressed. The choice of this latter threshold is based on the observation that in species with clear signature of translational selection, POC frequencies show little variation in genes below the median expression level (\hyperref[fig:CU4]{Fig. 4B}; \hyperref[suppfig:CU9]{Supplementary Fig. 9}). 

If constraints on synonymous codon usage are similar across species (\textit{i.e.} if $s^{hx}$ is constant), then $S^{hx}$ is expected to vary linearly with the effective population size ($S^{hx}=4\textit{N}_{\text{e}}s^{hx}$). To test this prediction, we sought to estimate \textit{N}$_{\text{e}}$ for each species. Lynch and colleagues recently compiled a list of species for which the germline mutation rate ($\mu$) and the level of neutral diversity ($\pi_s$) have been measured, and hence for which it is possible to infer the effective population size ($N_{e}=\pi_{s}/4\mu$) \citep{lynch_divergence_2023}. This list included 24 species of our data set, and in addition allowed us to get a proxy of \textit{N}$_{\text{e}}$ for 17 species, for which species from the same genus were available. To explore the relationship between $S^{hx}$ and \textit{N}$_{\text{e}}$ in more species, we also used three indirect proxies (longevity, body length and the ${dN}/{dS}$ ratio) that correlate with the effective population size (\hyperref[suppfig:CU10]{Supplementary Fig. 10}). 

Among the 223 species analyzed, the strongest intensity of selection is observed in the nematodes \textit{C. elegans} ($S^{hx}=1.3$) and \textit{C. nigoni} ($S^{hx} =1.0$; \hyperref[fig:CU7]{Fig. 7}). Dipters also show relatively strong values of $S^{hx}$ (mean = 0.63±0.16 sd), followed by lepidopters (mean $S^{hx}$ = 0.41±0.15 sd). In vertebrates, signals of translational selection are weak (mean = 0.15±0.12 sd), but nevertheless, $S^{hx}$ are on average significantly non null (Student's t-Test, p-value $<10^{-16}$). Our estimates match with those previously published for \textit{C. elegans}, \textit{Drosophila melanogaster}, human and mice \citep{dos_reis_estimating_2009}. As predicted by the drift barrier model, the species with the strongest signs of translational selection all show a relatively short lifespan, low body mass and low dN/dS (\hyperref[fig:CU7]{Fig. 7B,D}), \textit{i.e.} traits associated to organisms with large \textit{N}$_{\text{e}}$. Conversely species with traits associated to low \textit{N}$_{\text{e}}$ all show low $S^{hx}$. However, the correlations between $S^{hx}$ and \textit{N}$_{\text{e}}$ proxies are weak, and significant for only two of them (longevity and dN/dS) (\hyperref[fig:CU7]{Fig. 7B,D}). The weakness of these correlation might be due to the fact that these traits are only indirect proxies of \textit{N}$_{\text{e}}$. However, even for the few species for which it is possible to get more direct estimates of \textit{N}$_{\text{e}}$ (based on $\pi_s$ and $\mu$), the correlation between $S^{hx}$ and \textit{N}$_{\text{e}}$ remains weak (\hyperref[fig:CU7]{Fig. 7A}). 

\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=\textwidth]{Figure7.pdf}
    \end{center}                                                                       
    \caption[Relationship between \textit{N}$_{\text{e}}$ and translational selection intensity ($S$)]{\textbf{Relationship between \textit{N}$_{\text{e}}$ and translational selection intensity ($S$).} 
    Relationship between the population-scaled selection coefficient (\textit{S}) and \textit{N}$_{\text{e}}$ (\textbf{A}), longevity (days, log scale; \textbf{B}), body length (cm, log scale; \textbf{C}), dN/dS (log scale; \textbf{D}). The translational selection intensity $S$ is measured on the top 2\% most highly expressed genes ($S^{hx}$). Pagel's \textit{lambda} model is used to take into account the phylogenetic structure of the data in a regression model (the regression line is displayed in black when the correlation is significant). Error bars represent the 2.5th and 97.5th percentiles of $S$ values obtained from 1000 draws with replacement among the top 2\% most highly expressed genes, and the 50\% least expressed.
    }
    \label{fig:CU7}
\end{figure*}


\subsection{In species subject to translational selection, the tRNA pool evolves in response to changes in neutral substitution patterns}

The above analyses show that for most metazoan species, translational selection is very weak, and hence that their synonymous codon usage is essentially shaped by neutral neutral substitution patterns (NSP). Interestingly, even in species with clear signal of translational selection, codon usage appears to be influenced by variations in NSP. Notably, Diptera and Lepidoptera span a wide range of GC-content in non-coding regions (genome-wide average GCi ranging from 0.25 to 0.43), that strongly correlates with their average GC3 (from 0.32 to 0.71; \hyperref[fig:CU1]{Fig. 1B}). Yet, most of Diptera and Lepidoptera show a strong translational selection compared to other metazoans (26 out of 27 species with $S^{hx}>$ 0.3, the only exception being \textit{Eumeta japonica}, with $S^{hx}$ = 0.09). This raises the question of how the tRNA pool evolved in these species in response to NSP changes. To investigate this point, we focused our analyses on the 26 Diptera and Lepidoptera species with a strong signal of translational selection.

In this dataset, we observed that the decoding of 11 NNA/NNG synonymous codon pairs (Glu, Gln, Lys, Val, Ala, Pro, Thr, Ser, both CTA/CTG and TTA/TTG pairs of Leu, and the AGA/AGG 'duet' of Arg) never involves wobble pairing: the two complementary isodecoder tRNAs (anticodons UNN and CNN, respectively) are systematically present altogether in their genome (\hyperref[suppfig:CU12]{Supplementary Fig. 12}). Thus, for each of these 11 pairs, we identified the 'preferred' isodecoder tRNA (\textit{i.e.} the one with the highest gene copy number) in each species. We observed that the proportion of the 11 preferred tRNAs having a CNN anticodon in a given species correlates positively with the GCi of species (\hyperref[fig:CU8]{Fig. 8A}). This implies that tRNA gene copy number co-evolved in response to changes in NSP, consistent with the hypothesis that tRNA abundance is under selective pressure to match the demand in synonymous codon usage. For the two other synonymous codon pairs CGG/CGA of Arg and GGG/GGA of Gly, the CNN-tRNA is absent in 67\% of cases, where UNN-tRNA decodes both codons NNA/NNG (\hyperref[suppfig:CU12]{Supplementary Fig. 12}). 

NNT/NNC synonymous codon pairs (N=16) are generally decoded by a single isodecoder tRNA (\hyperref[fig:CU3]{Fig. 3B}, \hyperref[suppfig:CU2]{Supplementary Fig. 2}). Among dipters and lepidopters, this is the case for 94\% of the 416 NNT/NNC synonymous codons pairs analyzed (16 pairs $\times$ 26 species; \hyperref[suppfig:CU12]{Supplementary Fig. 12}). In such cases, shifts in NSP cannot be compensated for by a change in the relative abundance of isodecoder tRNAs. Nevertheless, the affinity of tRNAs for their cognate codons can be changed by post-transcriptional modifications, and hence might evolve in response to the demand. To investigate whether such changes occur, it is necessary to identify which of the two codons is best decoded by this unique isodecoder tRNA. For this, we relied on the fact that in species that are subject to translational selection, codons that are more efficiently decoded show a higher prevalence in highly expressed genes compared to lowly expressed ones (these codons will hereafter be referred to as preferred codons). Two sets of NNT/NNC synonymous codons pairs can be distinguished: the 7 pairs corresponding to the amino acids with duet codons (Phe, Cys, Tyr, Asp, His, Asn and the AGT/AGC 'duet' of Ser), and the 9 pairs from amino acids with triplet (Ile) or quartet codons (Val, Gly, Ala, Pro, Thr, Leu, Arg, Ser). For NNT/NNC duets, when a single tRNA is present (95\% of cases), it is always the GNN-tRNA, and in 99\% of cases it is the NNC codon, decoded through Watson-Crick pairing that is preferred. For 8 of the 9 other pairs, when a single tRNA is present (94\% of cases), it is always the ANN-tRNA, the only exception being Gly (GNN-tRNA). For Gly, the GGT codon, decoded via wobble pairing, is preferred to the GGC codon in 84\% of species. For the other pairs (decoded by ANN-tRNA) there is more variability: the NNC codon (wobble pairing) is preferred in 79\% of species, whereas the NNT codon (watson-crick pairing) is preferred in the others. These observations indicate that when a single tRNA is present for two codons, it is not systematically the one with watson-crick pairing that is the most efficiently translated. Furthermore, although the NNC codon tends to be preferred to the NNT codon (except for Gly), there are some variation across species, notably for those decoded by a ANN-tRNA (\hyperref[suppfig:CU11]{Supplementary Fig. 11}). We computed in each species the proportion of NNC preferred codons among NNT/NNC synonymous codon pairs decoded by a single tRNA. Interestingly, the species showing the highest proportion of NNT preferred codons are the ones with the lowest genomic GC content (\hyperref[fig:CU8]{Fig. 8B}). Thus, it appears that the relative affinity of ANN-tRNAs for the NNT or NNC codon can evolve in response to the demand. 

These observations suggest a straightforward model to explain variation in the set of optimal synonymous codons across species (\hyperref[fig:CU8]{Fig. 8C}). First, variation in mutational patterns or in the intensity of gBGC can lead to changes in the base composition of genomes, thereby directly shifting the codon usage of genes. Given that translational selection is a weak force, most genes are affected by this shift. This results in a change in the codon demand, and hence induces a selective pressure to change the pool of tRNA (both in terms of abundance and of affinity for their cognate codons). In turn, translational selection will modify the codon usage of highly expressed genes to match the new set of tRNAs, thereby reinforcing the selection on the tRNA pool to match the codon demand, and resulting to a new co-adaptation between the tRNA pool and codon usage.


\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=0.9\textwidth]{Figure8.pdf}
    \end{center}                                                                       
    \caption[Genomic substitution pattern shapes the tRNA pool]{\textbf{Genomic substitution pattern shapes the tRNA pool.} \textbf{A}: Relashionship between the \textit{per} species CNN fraction of preferred isodecoder tRNAs (corresponding to the most abundant tRNAs) among 11 NNA/NNG synonymous codon pairs and the gene average GC in introns (GCi), for Diptera (dark red) and Lepidoptera (light red). \textbf{B}: Relashionship between the \textit{per} species proportion of NNC preferred codons (the most overused codons in highly expressed genes compare to lowly expressed genes) among NNT/NNC synonymous codon pairs decoded by a single tRNA, along with the GCi. \textbf{A,B}: Pagel's \textit{lambda} model is used to take into account the phylogenetic structure of the data in a regression model (black line if significant). \textbf{C}: Hypothetical schemes explaining how synonymous codon usage can be shaped conjointly by translational selection and by neutral substitution patterns.}
    \label{fig:CU8}
\end{figure*}


\subsection{Weak translational selection in species with large intra-genomic variability in neutral substitution patterns}

An implicit assumption of the above model is that all genes of a given genome are affected by similar neutral substitution patterns. There is evidence however that some genomes are subject to heterogenous neutral substitution patterns. Notably, in mammals and birds, variation in recombination rates along chromosomes induce a strong heterogeneity in GC-content, driven by gBGC ~\citep{duret_biased_2009}. This process accounts for 70\% of the variance in synonymous codon usage among human genes \citep{pouyet_recombination_2017}. Thus, in these species, the synonymous codon usage of a given gene essentially depends on the base composition of the genomic region where it resides, as shown by the strong correlation observed between GC3 and GCi across human genes (\hyperref[fig:CU1]{Fig. 1C}). It is important to notice that these regional variations in GC-content affect all genes, even those that are widely expressed. To illustrate this point, we analyzed the codon usage of 2,249 human housekeeping genes (defined as genes that are in the top 20\% most highly expressed genes in at least 75\% of tissues). The distribution of GC3 in housekeeping genes shows a very strong heterogeneity (GC3 ranging from 25\% to 95\%; \hyperref[fig:CU9]{Fig. 9B}), as strong as in the entire gene set (\hyperref[fig:CU9]{Fig. 9A}). Housekeeping genes are involved in basal function and have to be expressed at high level in most cell types. This implies that in any given cell, there is a strong heterogeneity of the codon demand. Such a situation is predicted to hinder the co-adaptation between the tRNA pool and codon usage: any increase in the abundance of a given tRNA (say decoding a GC-ending codon) is expected to be beneficial for the translation of GC-rich genes, but detrimental for the translation of the GC-poor ones (and vice versa for a tRNA decoding an AU-ending codon). Hence, the selective pressure imposed by the codon demand is expected to maintain a balanced tRNA pool, able to decode GC-rich genes as well as GC-poor genes. In turn, the presence of a balanced tRNA pool should reduce the difference in translational efficiency between synonymous codons, and hence is expected to decrease the intensity of translational selection. Thus, genomes that are subject to heterogenous neutral substitution patterns are expected to be less subject to translational selection.
To test this prediction, we analyzed the relationship between the intensity of translational selection ($S^{hx}$) and the intra-genomic heterogeneity in base composition (assessed by the variance in GCi across genes). We observed that all species with a strong signal of translational selection show a very small variance in GCi, while species with a high variance in GCi show relatively low $S^{hx}$ (\hyperref[fig:CU9]{Fig. 9C}). This is consistent with the hypothesis that intra-genomic heterogeneity in base composition precludes translational selection. However, species with a high variance in GCi mainly correspond to three clades (Mammal, Aves, Hymenoptera) that have relatively small \textit{N}$_{\text{e}}$, and hence it is difficult to disentangle the impact of intra-genomic heterogeneity in base composition from that of drift (\hyperref[fig:CU9]{Fig. 9D}). 
In any case, even though intra-genomic heterogeneity in base composition might explain the weakness of translational selection in some species, there must be some other factors that affect the intensity of translational selection. Indeed, among insect species predicted to have a \textit{N}$_{\text{e}}$ similar to that of dipters, many show a low $S^{hx}$ despite a small variance in GCi (\hyperref[fig:CU9]{Fig. 9E,F}). 


\begin{figure*}[t]   
    \begin{center}
        \includegraphics[width=0.7\textwidth]{Figure9.pdf}
    \end{center}                                                                       
    \caption[Large intra-genomic variability in neutral substitution patterns impact on translational selection]{\textbf{Large intra-genomic variability in neutral substitution patterns impact on translational selection.} \textbf{A}: Distribution of GC content at the third position of codons (GC3) across human genes. \textbf{B}: Distribution of GC3 across human housekeeping genes (identified based on gene expression data from 27 healthy tissues, extracted from~\citep{pouyet_recombination_2017}. \textbf{C,D}: Relation between translational selection intensity \textit{S} and the gene GCi variance. \textbf{D}: Species are colored with a longevity gradient (log scale). \textbf{E,F}: Relation between translational selection intensity \textit{S} and longevity (days, log scale). \textbf{F}: Species are colored with a GC intron variance gradient (log scale).}
    \label{fig:CU9}
\end{figure*}




\section{Discussion}

\subsection{Predicting translationally optimal codons}

Patterns of SCU vary widely across metazoan species, and are strongly correlated to the base composition of their non-coding regions (\hyperref[fig:CU1]{Fig. 1A}). This implies that variation in codon usage across species are primarily shaped by differences in genome-wide neutral substitution patterns (driven by the underlying mutation pattern, by gBGC or both). NSPs vary not only across species, but also along chromosomes, and in some clades, such as tetrapods or hymenopters, this intra-genomic heterogeneity of NSPs is a major determinant of the variance in SCU among genes (\hyperref[suppfig:CU1]{Supplementary Fig. 1B}). The fact that genome-wide patterns of SCU are strongly affected by NSPs does not exclude that it can also be shaped by a selective pressure favoring the use of translationally optimal codons. Indeed, in dipters and lepidopters, which both show clear evidence of translational selection, we observed that the tRNA pool evolves in response to changes in genome-wide NSPs (\hyperref[fig:CU8]{Fig. 8}). Thus, variation in NSPs can lead to shifts in the translation apparatus, and thereby drive the evolution of SCU, not only in weakly expressed genes where codon usage is effectively neutral, but also in genes under strong translational selection. In other words, selective and non-adaptive models are not mutually exclusive, but it is important to take NSPs into account to be able to detect signatures of translational selection within genomes.

To quantify the intensity of translational selection in metazoans, we used a method based on standard population genetics equations, that infers the population-scaled selection coefficient ($S=4\textit{N}_{\text{e}}s$) from the difference in optimal codon frequency between highly expressed genes and weakly expressed genes \citep{sharp_variation_2005, dos_reis_estimating_2009}. This method first requires to identify the set of optimal codons in each species. To predict optimal codons, previous studies generally searched for codons whose frequency increases with gene expression level (\textit{e.g.} \citet{duret_expression_1999, dos_reis_estimating_2009}). One caveat, is that in some species, NSP varies with gene expression \citep{pouyet_recombination_2017}, which may therefore lead to errors in the inference of optimal codons. Furthermore, in that situation, the method would systematically overestimate $S$ for codons that are favored by NSPs in highly expressed genes. To limit this bias, we sought to predict optimal codons from the tRNA pool available in each species. For this, we estimated the abundance of each tRNA based on its gene copy number in the genome. In most species, we observed a strong correlation between the number of iso-acceptor tRNA gene copies and the frequency of their cognate amino-acid in the proteome (\hyperref[fig:CU3]{Fig. 3B}). These strong correlations are consistent with the fact that cellular tRNA abundance is highly constrained to match the amino-acid demand, and indicate that tRNA gene copy number is a good proxy to infer tRNA abundance, in agreement with previous experimental evidence from a limited set of species \citep{behrens_high-resolution_2021}.
Based on our estimates of the tRNA pool, we predicted two sets of putative optimal codons (POCs): for amino acids for which more than one iso-decoder tRNA is available, optimal synonymous codons were defined as those decoded by the most abundant tRNA (POC1); for amino acids encoded by NNC/NNU duet codons and with one single iso-decoder tRNA (GNN), the NNC codons were predicted to be optimal (POC2), based on previous studies showing that the wobble pairing GNN:NNU was less efficient than the Watson-Crick pairing GNN:NNC (\citet{stadler_wobble_2011, chan_frequent_2017}; \hyperref[fig:CU3]{Fig. 3B}).

Several lines of evidence indicate that our predictions of translationally optimal codons are accurate. First, our sets of POCs are consistent with previous predictions: we identified 25 POCs in \textit{C. elegans} and 27 in \textit{D. melanogaster}, while respectively 26 and 25 optimal codons had been inferred based on difference in codon usage between highly and lowly expressed genes \citep{duret_expression_1999}, of which 88.4\% and 88.0\% match with our POCs. Furthermore, the analysis of substitution patterns and polymorphism in \textit{Drosophila melanogaster} confirmed that selection favors POC alleles over non-POC alleles in highly expressed genes (\hyperref[fig:CU6]{Fig. 6}). Finally, we observed that although the definition of POC1 and POC2 relies on very different principles, the two sets of codons show very similar signatures of translational selection (\hyperref[fig:CU4]{Fig. 4C}). 

\subsection{Variation in the intensity of selection in favor of translationally optimal codons across metazoans}

For each species, we measured the frequency of optimal codons (combining POC1 and POC2) in highly expressed genes (top 2\%), to estimate the population-scaled selection coefficient in favor of translationally optimal codons ($S=4\textit{N}_{\text{e}}s$), using weakly expressed genes as a reference to account for the NSP \citep{sharp_variation_2005}. Across the 223 species, the highest values of $S$ are observed in Caenorhabditis nematodes ($S=1.25$ in \textit{C. elegans} and $S=1.00$ in \textit{C. nigoni}). We also found a clear signal of translational selection in diptera (mean $S=0.63$, N=19 species), and to a lesser extent in lepidoptera (mean $S=0.41$, N=8 species). Overall, estimates of $S$ are weaker in other clades (\hyperref[fig:CU7]{Fig. 7}). The weakness of translational selection in vertebrates (mean $S=0.16$, N=100 species) was a priori expected given that these organisms tend to have relatively small \textit{N}$_{\text{e}}$. But besides Caenorhabditis and dipters, our dataset included 83 invertebrate species covering a wide range of clades (58 other insects, 12 other Ecdysozoa, 6 Spiralia, 4 Cnidaria, 3 Deuterostomia). What is surprising is that all these species show $S$ values that are lower than the average of dipters. This implies that the high values of $S$ observed in Caenorhabditis and in dipters represent exceptions rather than the rule, and that translational selection is weak in most metazoan lineages.

If the selection coefficient in favor of optimal codons ($s$) was constant across metazoans, then S should scale linearly with \textit{N}$_{\text{e}}$. To test this prediction, we used silent-site polymorphism and germline mutation rate data \citep{lynch_divergence_2023} to estimate the effective population size ($N_{e}=\pi_{s}/4\mu$, hereafter noted \textit{N}$_{\text{e}}^{\pi\mu}$) in 41 species. As expected $S$ tends to increase with \textit{N}$_{\text{e}}^{\pi\mu}$, but the correlation is not significant after accounting for phylogenetic inertia (\hyperref[fig:CU7]{Fig. 7A}). The weakness of the correlation might be due to the fact that these two parameters evolve on different time scales: \textit{N}$_{\text{e}}^{\pi\mu}$ is indicative of the recent effective population size (on the order of \textit{N}$_{\text{e}}$ generations) and hence can change quite rapidly compared to $S$, that is estimated from the codon composition of genomes, resulting from a long-term accumulation of substitutions. This can explain why \textit{C. nigoni} and \textit{C. elegans} display similar values of $S$, despite a 75-fold difference in \textit{N}$_{\text{e}}^{\pi\mu}$ (respectively \textit{N}$_{\text{e}}^{\pi\mu} =9.4\times 10^{6}$ and \textit{N}$_{\text{e}}^{\pi\mu}=1.2\times 10^{5}$). This difference in \textit{N}$_{\text{e}}^{\pi\mu}$ is due to the fact that \textit{C. nigoni} is an outcrossing species, like most other Caenorhabditis species, while the \textit{C. elegans} lineage evolved towards selfing hermaphroditism \citep{li_specialist_2014, vielle_convergent_2016}. This transition in reproductive mode is recent, and hence the SCU of \textit{C. elegans} still retains the signature of strong translational selection inherited from its outcrossing ancestors. Thus, we can predict that the SCU of \textit{C. elegans} is not at selection-mutation-drift equilibrium (which could be tested by analyzing synonymous polymorphism). 

To further test the relationship between $S$ and \textit{N}$_{\text{e}}$, we considered three parameters (longevity, body length and dN/dS), that are all correlated with \textit{N}$_{\text{e}}^{\pi\mu}$ (\hyperref[suppfig:CU10]{Supplementary Fig. 10}), but that are expected to reflect \textit{N}$_{\text{e}}$ over a longer time scale. A further interest of these proxies is that they can be estimated on much larger datasets (150 to 223 species). But here again we obtained similar results: $S$ tends to increase with \textit{N}$_{\text{e}}$, but correlations are weak, marginally significant after accounting for phylogeny (\hyperref[fig:CU7]{Fig. 7B,C and D}). The weakness of the correlation is mainly due to the fact that some species have a low $S$, despite life-history traits or dN/dS values indicative of a high \textit{N}$_{\text{e}}$.

Not only the correlations between \textit{N}$_{\text{e}}$ and $S$ are weak, but also the range of variation in $S$ appears to be quite limited compared to what would be expected given the variance in \textit{N}$_{\text{e}}$.
For instance, the mean value of \textit{N}$_{\text{e}}^{\pi\mu}$ is about 15 times higher in diptera than in mammalia (based on respectively 6 and 41 species for which \textit{N}$_{\text{e}}^{\pi\mu}$ can be estimated; \citet{lynch_divergence_2023}). Yet, the mean value of $S$ is only 5.3 times higher in diptera (mean $S=0.63$, N=19 species) than in mammals (mean $S=0.12$, N=65 species). Thus, the difference in $S$ between diptera and mammals is smaller than what would be expected if $S$ scaled linearly with \textit{N}$_{\text{e}}$.

One possible explanation for this discrepancy is that $S$ is overestimated in mammals. As discussed by \citet{dos_reis_estimating_2009}, the estimate of $S$ is based on the assumption that NSPs are constant across genes, \textit{i.e.} that the difference in optimal codon frequency between highly and weakly expressed genes is entirely due to translational selection. In reality, in some species, NSPs vary with gene expression (\textit{e.g.} in humans \citet{pouyet_recombination_2017}). To try to account for these variations, we measured the differences in POC frequency between highly and weakly expressed genes, controlling for differences in the corresponding triplet frequencies in introns. It is however possible that the base composition of introns is not a perfect predictor of NSPs, notably because introns are affected by indels and transposable elements, which are not allowed in coding regions. This is well illustrated by POC2 codons in humans, whose frequency clearly covaries with their non-coding controls, but with a wider amplitude in exons than in introns (\hyperref[fig:CU4]{Fig. 4A}).

An alternative hypothesis is that the discrepancy might result from a strong heterogeneity in the fitness effect of synonymous mutations. Indeed, the analysis of synonymous polymorphism in \textit{D. melanogaster} indicated that a majority of codons are under weak selection in favor of translationally optimal codon ($|$\textit{N}$_{\text{e}} s|\approx1$), but that a small fraction (10\%-20\%) are under strong selection ($|$\textit{N}$_{\text{e}} s|>10$; \citet{machado_pervasive_2020}. With a \textit{N}$_{\text{e}}$ value 15 times lower, the first class of codons is expected to evolve neutrally in mammals. But the second class of codons would still appear under effective translational selection, which might explain the small but non-null value of $S$ measured in mammals.

One last unexpected observation is that many species predicted to have a high \textit{N}$_{\text{e}}$ (based on their LHTs or dN/dS) show very weak $S$ (\hyperref[fig:CU7]{Fig. 7B,C and D}). In some species, this could be explained by the heterogeneity of NSPs along chromosomes, inducing a strong variance in SCU that precludes a co-adaptation of the tRNA pool. This might be the case notably for some hymenopters, which, like tetrapods, are subject to gBGC \citep{wallberg_extreme_2015} and present a very strong heterogeneity in NSPs (\hyperref[fig:CU9]{Fig. 9C}). However, our dataset also includes some species with small $S$ values, despite a high \textit{N}$_{\text{e}}$ proxy and homogenous NSPs. So finally, we are left with the conclusion that variation in $S$ across metazoans are not driven simply by the drift barrier and by gBGC, but that they are also probably due to variation in s, the selection coefficient in favor of translationally optimal codons. There is evidence that in unicellular organisms, the selective force for the optimization of SCU is the maximization of cellular growth \citep{rocha_codon_2004, sharp_variation_2005}. It is possible that the selective pressure on cellular growth also vary across metazoans. Most Caenorhabditis species grow in ephemeral environments (rotting vegetation) and hence have been selected for their capacity to proliferate very rapidly \citep{cutter_caenorhabditis_2015}. \citet{manthey_rapid_2024} recently quantified growth rates in 33 insects. The only dipter present in their dataset (\textit{Lucilia sericata}) is the species that presented the highest growth rate, 12 times higher than the average growth rate of other holometabole insects (N=10) and 52 times higher than the average growth rate of hemimetabole insects (N=22). If the \textit{Lucilia sericata} is representative of other dipters, this might explain why translational selection is particularly strong in that clade compared to other insects. It is noteworthy that the two invertebrate species that have been historically used as model organisms (\textit{D. melanogaster} and \textit{C. elegans}) both belong to the very rare metazoan clades with clear evidence of translational selection. This might reflect the fact that they have been chosen a model organisms for the very reason that they can grow very fast in the lab.





\section{Materials \& Methods}
\label{sec:MaterialsMethodsCU}


\subsection{Gene expression and data collection}

The reference genome assemblies and genome annotations were acquired from the National Center for Biotechnology Information (NCBI; \citet{sayers_database_2022}.
We obtained gene expression data for 257 metazoan species from GTDrift~\citep{benitiere_gtdrift_2024}, where Fragment \textit{Per} Kilobase of exon \textit{per} Million mapped reads (FPKM) was estimated over thousands of RNA-seq samples using cufflinks. For each species we considered the \textit{per}-gene median FPKM values across all analyzed RNA-seq samples.
Additionally, a phylogenetic tree was retrieved from GTDrift to account for phylogenetic inertia \citep{benitiere_gtdrift_2024}.


\subsection{tRNAscan-SE annotation}

If for a given species tRNA genes copies were previously annotated and so present in the NCBI annotation file, we took these annotations into account (N=44 species). Other wise (N=213 species) we annotated tRNA gene copies using the program tRNAscan-SE 2.0.12 (Nov 2022), with the -E option specifically designed for eukaryotic tRNA identification search~\citep{chan_trnascan-se_2021}. To keep in the study functional gene copies we retained those with a score exceeding 55, threshold based on~\citet{chan_trnascan-se_2021} analysis. Thus, for each of these copies we obtained the decoded codon and the translated amino acid. 

% command: tRNAscan-SE -E -o tRNA.tab -S off file-out

\subsection{Codon usage}

For each species in our study, we conducted a detailed assessment of codon usage across the longest annotated coding sequences (CDS) of each expressed gene within our dataset. This analysis was paralleled by an examination of the occurrences of nucleotide triplets within intron regions. It is important to note that our analysis deliberately excluded the acceptor and donor splice sites to avoid skewing the results with these highly conserved motifs.


\subsection{Site constraint}

Multiple gene alignments of 976 BUSCO genes and 293 species were collected to study site constraints from the metazoa dataset alignment of GTDrift repository~\citep{benitiere_gtdrift_2024}. For each gene, we determined the proportion of gaps at each site across the alignment. This information guided the \textit{per}-species segmentation of genes into bins, with each bin representing 25\% of a gene sequence. Our examination of tetrapods focused on sites located beyond 1,000 base pairs downstream from the start codon.


\subsection{SNPs analysis}

We used polymorphism data from the \textit{Drosophila} Genetic Reference Panel 2 (DGRP2)~\citep{mackay_drosophila_2012, huang_natural_2014}, where polymorphic sites have been identified from comparisons across 205 inbred lines of \textit{Drosophila melanogaster}, downloaded from \url{http://dgrp2.gnets.ncsu.edu/data/website/dgrp2.vcf}. We converted the single nucleotide polymorphism (SNP) coordinates from the dm3 genome assembly to the dm6 assembly, with the liftOver utility~\citep{hinrichs_ucsc_2006} of the UCSC genome browser, using a whole genome alignment between the two assemblies downloaded from \url{https://hgdownload.soe.ucsc.edu/goldenPath/dm3/liftOver/dm3ToDm6.over.chain.gz}. We kept in the study 3,738,302 biallelic SNPs for which more than 181 individuals have been genotyped.

We then identified two sister species \textit{Drosophila simulans} and \textit{Drosophila erecta} that we aligned against \textit{Drosophila melanogaster} genome using liftOver utility \citep{hinrichs_ucsc_2006}. We removed from the analysis genes located in regions where the multiple alignment was of poor quality (\hyperref[fig:CU8]{Supplementary Fig. 8}. We used the program est-sfs release 2.04~\citep{keightley_inferring_2018} to polarize SNPs, \textit{i.e.} to identify the ancestral allele and the derived allele. 

For the longest annotated coding sequence of each expressed gene we were able to identify the ancestral and derived codons for each SNP. The same approach was applied for intron regions by studying nucleotide triplets.

To determine the confidence interval for each data point (\hyperref[fig:CU6]{Fig. 6}), we employed a simulation approach. We simulated a sample with a similar structure in terms of the number of sites \textit{per} gene and the SNP rate, using a binomial distribution. We calculated the average SNP rate for each simulation, repeating the process 100 times. Afterward, we utilized the 2.5th and 97.5th percentiles of these mean values to establish the error bars.

\subsection{Substitutions analysis}

Based on the multiple genome alignment of \textit{Drosophila simulans}, \textit{Drosophila erecta} and \textit{Drosophila melanogaster} previously described, we identified the non polymorphic sites where a substitution occurred. To do so, we studied sites for which we were able to determine the ancestral and substituted allele by considering the most parsimonious scenario :
if reference alleles of \textit{D.erecta} = \textit{D.melanogaster} or \textit{D.simulans} = \textit{D.melanogaster} there is no substitution;
if reference alleles of \textit{D.erecta} = \textit{D.simulans} but differ from \textit{D.melanogaster} then there is a substitution and the ancestral allele is the one observed on \textit{D.erecta} and \textit{D.simulans}.

We identified a total of 1,759,664 substitutions, and were able for each codon containing at least one substitution to determine its ancestral and substituted state. The same approach was applied for intron regions by studying nucleotide triplets. This protocol was executed on the longest annotated coding sequence of each expressed gene.

\subsection{Effective population size estimates}

We retrieved proxies for the effective population size from the GTDrift data resource~\citep{benitiere_gtdrift_2024}, which included life history traits such as body length, longevity, and the ratio of non-synonymous to synonymous substitutions rate (${dN}/{dS}$). It is expected that the genome-wide ${dN}/{dS}$ ratio increases during prolonged periods of small \textit{N}$_{\text{e}}$, attributed to the fixation of slightly deleterious mutations \citep{ohta_nearly_1992, galtier_adaptive_2016}. To enhance the dataset, we supplemented the effective population size proxies with body mass data extracted from \citet{lynch_divergence_2023} for 45 species, encompassing 26 species within our dataset and 19 for which species from the same genus were available.

Furthermore, from \citet{lynch_divergence_2023}, we obtained direct estimates of \textit{N}$_{\text{e}}$ by deriving the effective population size ($N_{e}=\pi_{s}/4\mu$) using the germline mutation rate ($\mu$) and the level of neutral diversity ($\pi_s$) for 45 species, comprising 27 species within our dataset and 18 for which species from the same genus were available. Additionally, we expanded our dataset with the \textit{N}$_{\text{e}}$ estimate for \textit{C. nigoni} by including $\pi_s=0.06$ (Asher Cutter, personal communication) and $\mu=1.3\times10^{-9}$ (\citet{denver_variation_2012}; assuming a similar mutation rate as in \textit{C. briggsae}). We calculated $N_{e}=\pi_s/4\mu\approx 1.1\times 10^{-7}$.

\subsection*{Acknowledgements}
Computational analyses were performed using the computing facilities of the CC LBBE/PRABI and the Core Cluster of the Institut Français de Bioinformatique (IFB) (ANR-11-INBS-0013).
Silhouette images of taxonomic Families originate from PhyloPic developed and maintained by Mike Keesey available at \url{https://www.phylopic.org/}.

\subsection*{Author contributions statement}
F.B. conceived the pipeline and conducted the analyses. F.B. drafted the manuscript. All authors reviewed the manuscript. 

\subsection*{Funding}
This work was funded by the French National Research Agency (ANR-20-CE02-0008-01 "NeGA").

\subsection*{Competing interests}
The authors declare no conflicts of interest.
